ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -818,6 +803,20 @@ private static void createReplImportTasks(
         throw new SemanticException(ErrorMsg.DATABASE_NOT_EXISTS.getMsg(tblDesc.getDatabaseName()));
       }
     }
+
+    if (table != null) {
+      if (!replicationSpec.allowReplacementInto(table)) {
+        // If the target table exists and is newer or same as current update based on repl.last.id, then just noop it.
+        return;
+      }
+    } else {
+      // If table doesn't exist, allow creating a new one only if the database state is older than the update.
+      if ((parentDb != null) && (!replicationSpec.allowReplacementInto(parentDb))) {
```

Comment:
We have 2 sets of methods. One is allowReplacementInto which checks the object's ID against current_state_id. There is another one allowEventReplcacementInto which checks it against event_id. Both are different. isOlderThan cannot replace both methods. So, need to find 2 appropriate methods. May be isOlderThan and isEventOlderThan, but that doesn't seem good names. So, just want to know if you have any better suggestions for these names.